### PR TITLE
feat: update Azure CNI version to 1.0.17

### LINKED ISSUE #408: Upgrade the Azure CNI version to v1.0.17
We, Azure Container Networking team have released a new version of Azure CNI: https://github.com/Azure/azure-container-networking/releases/tag/v1.0.17

Please pick this up in your next aks-engine release.

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**Reason for Change**:

https://github.com/Azure/azure-container-networking/releases/tag/v1.0.17

**Issue Fixed**:

Fixes #408 

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

TODO: 

- [x] Add binary to acsmirror

### CODE REVIEW

File: packer/install-dependencies.sh [hunk 1]
```diff
@@ -25,7 +25,7 @@ fi
 
 installClearContainersRuntime
 
-VNET_CNI_VERSIONS="1.0.14 1.0.15 1.0.16"
+VNET_CNI_VERSIONS="1.0.16 1.0.17"
```
Comment (CecileRobertMichon) [inline packer/install-dependencies.sh]: I removed the older versions, no need to keep around too many versions. Keeping 1.0.16 for easy revert so we don't have to re-publish the image if needed.

**Comment by jackfrancis** [comment]
@CecileRobertMichon FYI we manually curate Azure CNI artifacts behind acsmirror

**Comment by PatrickLang** [comment]
@jackfrancis - how did the soak tests go?

**Comment by CecileRobertMichon** [comment]
soak tests look good so far, will wait until EOD to merge

**Pull request merged**